lib/internal/Magento/Framework/View/Element/AbstractBlock.php

```diff
@@ -665,6 +665,15 @@ public function toHtml()
         }
         $html = $this->_afterToHtml($html);
 
+        /** @var \Magento\Framework\DataObject */
+        $transportObject = new \Magento\Framework\DataObject(
```

Comment:
There are at least 2 reasons not to trigger this event in AbstractBlock:

1. Abstract block is overloaded with behavior
2. If some block will just implement BlockInterface (not extend from AbstractBlock), this event will not be called.

Overall approach of Magento 2 is to use composition instead of inheritance to reuse behavior. In this case you need to reuse behavior common for all `toHtml()` calls. Please consider putting this behavior to  `\Magento\Framework\View\Layout::renderNonCachedElement`.